### PR TITLE
Skip `test_multi_gpu_data_parallel_forward` for `UperNetModelTest`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Same as `BEIT`, this model uses some `add_module` and we need to skip this test.

### CODE REVIEW

File: tests/models/upernet/test_modeling_upernet.py [hunk 1]
```diff
@@ -199,6 +199,11 @@ def test_save_load_fast_init_from_base(self):
     def test_save_load_fast_init_to_base(self):
         pass
 
+    @require_torch_multi_gpu
```
Comment (NielsRogge) [inline tests/models/upernet/test_modeling_upernet.py:202]: Why add this annotator if we skip the test?

**Comment by NielsRogge** [comment]
Wait, are you saying that models that using add_module can't leverage multi-GPU training?

File: tests/models/upernet/test_modeling_upernet.py (hunk 1 above)
Comment (ydshieh) [reply tests/models/upernet/test_modeling_upernet.py:202]: It's fine without it. But we just keep the original context to make it clear - as we do in many other places, like

```
    @unittest.skip(reason="`saved_model` doesn't work with nested outputs so no preparation happens.")
    @slow
```
or
```
    @unittest.skip("Requires an update of the env running those tests")
    @require_torch_multi_gpu
    @require_fairscale
```
or 
```
    @require_tf
    @unittest.skip("Image segmentation not implemented in TF")
```

**Comment by ydshieh** [comment]
> Wait, are you saying that models that using add_module can't leverage multi-GPU training?

Yes, that's my observation while debugging BEIT, and then with `LayoutLMV2`, `Data2VecVision` etc.

```
    @unittest.skip(
        reason="Data2VecVision has some layers using `add_module` which doesn't work well with `nn.DataParallel`"
    )
```

**Comment by ydshieh** [comment]
@NielsRogge Let me know if you have further question :-) before giving 👍 . Thank you 🙏

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing!

**Pull request merged**